modules/standard/Set.chpl

```diff
@@ -266,15 +266,22 @@ module Set {
     }
 
     /*
-      Add a copy of the element `x` to this set. Does nothing if this set
-      already contains an element equal to the value of `x`.
+      Add a copy of the element `element` to this set. Does nothing if this set
+      already contains an element equal to the value of `element`.
 
-      :arg x: The element to add to this set.
+      :arg element: The element to add to this set.
     */
-    proc ref add(in x: eltType) lifetime this < x {
+    proc ref add(in element: eltType) lifetime this < element {
 
-      // Remove `on this` block because it prevents copy elision of `x` when
-      // passed to `_addElem`. See #15808.
+      // Remove `on this` block because it prevents copy elision of `element`
+      // when passed to `_addElem`. See #15808.
+      _enter(); defer _leave();
+      _addElem(element);
+    }
+
+    pragma "last resort"
+    deprecated "The argument name `x` has been deprecated for function `add`, please use `element` instead"
+    proc ref add(in x: eltType) lifetime this < x {
       _enter(); defer _leave();
```

Comment:
The lifetime constraint may mean this doesn't work, but I tend to like having the deprecated function call the new function as much as possible.  This code is probably short enough that it is less likely but I worry about someone else coming in and modifying the wrong version of the function, and then having the changes not be reflected in the new version when the old version gets deleted

Reply:
It looks like that works just fine, I've changed the code to reflect that; thanks for the suggestion.